### ISSUE TITLE
Adding new "fighter" class to enable diversity of fighter types

### ISSUE THREAD

**Issue by TheMarksman-ES** [endless-sky/endless-sky#4033]
In reference to the discord, the idea is to add a new class of Fighters, tentatively named as "Gunships" to enable more players to choose to add them to their fleets. These Gunships would have their own docking port and be able to carry (tentatively) more weapons and/or engine space.

Tentatively, they'll have to be bigger than the current fighters, more likely to be the size of interceptors or bigger, but with no capabilities to jump to other systems.

This would also complement, if ever implemented, heavier fighter weaponry.

Anyways, if you have not voted, please go to the discord to vote. Thanks.

**Comment by Bladewood** [comment]
This sounds like it would erode the distinction between fighters and interceptors - at which point I question why bother having all three (four including drones) separate classes at all, especially since the main point of the latter is literally "fighter with a hyperdrive":
> Classified as an interceptor rather than a fighter because it has its own hyperdrive instead of needing to be carried inside a larger ship, the Tarazed Sparrow ...

**Comment by MCOfficer** [comment]
also, for the record, by classic definitions interceptors are usually a step down from gunships - not to mention the confusion this will cause with the navy gunship. you might want to work on that name a bit more.

**Comment by TheMarksman-ES** [comment]
>also, for the record, by classic definitions interceptors are usually a step down from gunships - not to mention the confusion this will cause with the navy gunship. you might want to work on that name a bit more.

I've been informed that the navy employ gunboats, not gunships

**Comment by Amazinite** [comment]
Let it be known that we literally have a freighter named Freighter.

**Comment by Nescio0** [comment]
The idea is nice, although it would require a bit of a code overhaul. Currently most is done with a boolean `isFighter`; 'false' means a drone, 'true' a fighter. Perhaps it would make sense to replace both the `isFighter` and the `canBeCarried` booleans with an integer `canBeCarried`, to allow for more types of bays and carried ships, e.g. 1 == drone, 2 == fighter, 3 == gunship, 4 == bomber (and 0 for ships capable of independent hyperspace travel). This also offers a more possibilities for plug-ins.

**Comment by Amazinite** [comment]
>This sounds like it would erode the distinction between fighters and interceptors - at which point I question why bother having all three (four including drones) separate classes at all, especially since the main point of the latter is literally "fighter with a hyperdrive":

Interceptors are already practically non-existent by the time we reach T2, so I see no issue in adding another class of carried ship that is larger than fighters and has its own specific bay type for an alien faction to use. Not every idea suggested need to be implemented by humans.

>Why not just create some bigger fighters?

Because bigger fighters can still fit into the same bays as smaller fighters. I somewhat jokingly referred to a gunship class as being [light warships that go into bays](https://github.com/endless-sky/endless-sky/pull/2200#issuecomment-283503934) in the past (seeing as how fighters are somewhat interceptors that go into bays, ignoring the fact that no fighter is actually better than any interceptor of the same tier), but at the very least a gunship would be more powerful enough than a fighter of the same tier that it would warrant its own category.

**Comment by TheMarksman-ES** [comment]
>@TheMarksman My idea is to add fighter weaponry, allowing fighters to become Capital ship killers.
The fighter type could have distinct classes according to what it's supposed to do.
Strike fighter — armed with 2–4 missile launchers, attacks point targets and small ships, once below 50% health or once the ammo is expended, goes to restock.
Gunship — armed with a turret (e. g. the Barb), see above.
Bomber — a fighter featuring 1–4 torpedo bays, it's task is to fire its ammo in max. four volleys and go home.

>Gunship and Bomber could be of a separate gunship type, larger than a usual fighter.
-Tadrix.

I'm rather keen of tadrix idea of adding different classes with different attributes for them, even if my idea and he's differ slightly.

**Comment by TheMarksman-ES** [comment]
But it's a start.

Anyways, to add on to his idea, my thoughts on this is that Strike fighters should take on anti fighter roles, choosing to attack other fighters and avoid enemy capital ships.

Gunships should still be larger than ordinary fighters, though they will also be allowed to have more weapons and outfit space, probably around the same as the sparrow, though that's just an estimate and template.

A new type of engine(X2050 Ion Engine)[if this isn't familiar, it's something I have planned for my own PR] should be given to these Gunships to bolster their speed and make them more effective at anti capital ship cum anti fighter roles.

Bombers should have "bombing bays" (entertaining the thought of allowing other classes of fighter's weapons space and gunports to be changed via the outfitters to become bomb bays) to enable these bombers to become a very op anti ship weaponry, though ammo will be the limiting factor preventing these from becoming too overpowered.

**Comment by TheMarksman-ES** [comment]
@Amazinite side note, would it be practical to make another issue regarding fighter weaponry, or wait on your poll?

**Comment by Pointedstick** [comment]
I don't see the need for a new class here. More actual Fighter hulls might be nice--or more real differences between the existing hulls (e.g. the Lance and Dagger are practically identical), but I don't see the value in creating a new class of fighter that would require its own bay type. The additional specificity and complication strikes me as something that would make people less likely to use fighters, not more.

Fighters already have enough customization (in fact, some suggest that it's too much). What's needed is easier maintenance and better playability.

**Comment by Amazinite** [comment]
In v0.9.15 with #5971 we made it so that which ships can be carried is configurable in the data instead of being hardcoded to just fighters and drones. If a plugin wishes to have a "gunship" category of ships then they can do so, but I don't think we should be aiming to add such a category to vanilla at this time.
